aptos-move/move-examples/post_mint_reveal_nft/sources/minting.move

```diff
@@ -0,0 +1,861 @@
+        nft_claimer: &signer,
+        amount: u64
+    ) acquires NFTMintConfig, PublicMintConfig, WhitelistMintConfig, CollectionConfig, SourceToken {
+        assert!(exists<CollectionConfig>(@post_mint_reveal_nft) && exists<WhitelistMintConfig>(@post_mint_reveal_nft) && exists<PublicMintConfig>(@post_mint_reveal_nft), error::permission_denied(ECONFIG_NOT_INITIALIZED));
+
+        let whitelist_mint_config = borrow_global_mut<WhitelistMintConfig>(@post_mint_reveal_nft);
+        let public_mint_config = borrow_global<PublicMintConfig>(@post_mint_reveal_nft);
+
+        let now = timestamp::now_seconds();
+        assert!((now > whitelist_mint_config.whitelist_minting_start_time && now < whitelist_mint_config.whitelist_minting_end_time) || (now > public_mint_config.public_minting_start_time && now < public_mint_config.public_minting_end_time), error::permission_denied(EMINTING_IS_NOT_ENABLED));
+
+        let nft_mint_config = borrow_global<NFTMintConfig>(@post_mint_reveal_nft);
+        let collection_config = borrow_global_mut<CollectionConfig>(@post_mint_reveal_nft);
+        let resource_signer = create_signer_with_capability(&nft_mint_config.signer_cap);
+        if (!exists<SourceToken>(@post_mint_reveal_nft)) {
+            // This is the collection that contains certificate tokens, which will allow users to exchange to an NFT in the destination collection post-mint.
+            let source_collection_name = collection_config.collection_name;
+            string::append_utf8(&mut source_collection_name, b" - source collection");
+            create_collection(&resource_signer, source_collection_name, collection_config.collection_description, utf8(b""), collection_config.collection_maximum, collection_config.collection_mutate_config);
+
+            let source_token_data_id = create_tokendata(
+                &resource_signer,
+                source_collection_name,
+                // token name
+                utf8(b""),
+                // token description
+                utf8(b""),
+                // token maximum is equal to the number of tokens in the destination collection
+                vector::length(&collection_config.token_uri),
+                // token uri
+                utf8(b""),
+                signer::address_of(&resource_signer),
+                1,
+                0,
+                create_token_mutability_config(
+                    &vector<bool>[ false, false, false, false, true ]
+                ),
+                vector<String>[string::utf8(BURNABLE_BY_OWNER)],
+                vector<vector<u8>>[bcs::to_bytes<bool>(&true)],
+                vector<String>[string::utf8(b"bool")],
+            );
+
+            move_to(&resource_signer, SourceToken { token_data_id: source_token_data_id });
+        };
+
+        // if this is the whitelist minting time
+        if (now > whitelist_mint_config.whitelist_minting_start_time && now < whitelist_mint_config.whitelist_minting_end_time) {
```

Comment:
Isn't this redundant with line 276?

Reply:
line 276 is going to abort if this is neither the whitelist minting time nor the public minting time.

the if statement here covers whitelist minting time, and the else statement here covers public minting time (since if it's neither it would have aborted earlier.

Reply:
public minting time should be strictly > whitelist minting time. If line 276 doesn't abort, that's mean current time >= whitelist minting time. This renders this statement unnecessary. Or did I misunderstand something somewhere?

Reply:
ah yeah you are right, will update this